File: examples/go/core/shared/app_infra_contracts.go
package shared

type AppHttpClient interface {
	Request(config any) (any, error)
}

type AppStorageClient interface {
	Get(key string) (any, error)
	Set(key string, value any) error
}

type AppCache interface {
	Get(key string) (any, error)
	Set(key string, value any, ttlSeconds int) error
}

type AppEventPublisher interface {
	Publish(event string, payload any) error
}

File: examples/java/core/shared/AppInfraContracts.java
package core.shared;

public final class AppInfraContracts {
  private AppInfraContracts() {}

  public interface AppHttpClient {
    Object request(Object config) throws Exception;
  }

  public interface AppStorageClient {
    Object get(String key) throws Exception;
    void set(String key, Object value) throws Exception;
  }

  public interface AppCache {
    Object get(String key) throws Exception;
    void set(String key, Object value, Integer ttlSeconds) throws Exception;
  }

  public interface AppEventPublisher {
    void publish(String event, Object payload) throws Exception;
  }
}
